index.js

```diff
@@ -6,6 +6,6 @@
  */
 
 module.exports = function isGitUrl(str) {
-  var re = /(?:git|ssh|https?|git@[\w\.]+):(?:\/\/)?[\w\.@:\/~_-]+\.git(?:\/?|\#[\d\w\.\-_]+?)$/;
+  var re = /^(?:git|ssh|https?|[\d\w\.\-_]+@[\w\.]+):(?:\/\/)?[\w\.@:\/~_-]+(?:\/?|\#[\d\w\.\-_]+?)$/;
```

Comment:
I think we need to add `git` back in the regex in both places,. I'd rather us be more explicit about the formats being accepted, since this is supposed to check for `.git` urls. I'm okay with doing something like `https?|(git|something else)@`. 

Let's just list out the formats that need to be accepted, and maybe add failing unit tests, then we can rewrite the regex to work for those. how does that sound?